packages/contracts/contracts/Greenproof.sol

```diff
@@ -137,6 +196,31 @@ contract Greenproof is SolidStateDiamond {
         );
     }
 
+    fallback() external payable override(IProxy, Proxy) {
+        if (_isContractPaused) {
+            revert PausedContract();
+        }
+        address implementation = _getImplementation();
+
+        if (!implementation.isContract()) {
+            revert ProxyError("implementation must be contract");
+        }
+
+        assembly {
+            calldatacopy(0, 0, calldatasize())
+            let result := delegatecall(gas(), implementation, 0, calldatasize(), 0, 0)
+            returndatacopy(0, 0, returndatasize())
+
+            switch result
+            case 0 {
+                revert(0, returndatasize())
+            }
+            default {
+                return(0, returndatasize())
+            }
+        }
+    }
```

Comment:
I don't understand it. You have to either comment it, link to explanation or ask somebody else to review it 😄